crates/p2p/src/swarm/setup/events.rs

```diff
@@ -0,0 +1,50 @@
+//! Event types for the setup protocol.
+//!
+//! This module defines the events that are emitted during the setup phase
+//! of peer-to-peer connections, providing information about handshake progress
+//! and key exchange completion.
+
+use libp2p::{PeerId, identity::PublicKey};
+
+use crate::swarm::errors::SetupError;
+
+/// Events emitted during the setup phase of peer connections.
+///
+/// These events provide feedback on the progress of the handshake protocol
+/// and are used to communicate setup milestones to the swarm.
+#[derive(Debug)]
+pub enum SetupBehaviourEvent {
+    /// Indicates that an application public key has been received from a peer.
+    ///
+    /// This event is fired when the local node successfully receives and
+    /// processes a peer's application public key during the handshake.
+    AppKeyReceived {
+        transport_id: PeerId,
+        app_public_key: PublicKey,
+    },
+
+    /// Indicates that signature verification failed.
+    ///
+    /// This event is fired when the signature verification fails for a peer's
+    /// setup message, indicating the connection should be dropped.
+    ErrorDuringSetupHandshake {
+        transport_id: PeerId,
+        error: SetupError,
+    },
+}
+
+/// Events emitted during the setup phase of peer connections.
+///
+/// These events provide feedback on the progress of the handshake protocol
+/// and are used to communicate setup milestones to the swarm.
+#[derive(Debug)]
+pub enum SetupHandlerEvent {
+    /// Indicates that an application public key has been received from a peer.
+    ///
+    /// This event is fired when the local node successfully receives and
+    /// processes a peer's application public key during the handshake.
+    AppKeyReceived { app_public_key: PublicKey },
+
+    /// Something has failed during setup handshake.
+    ErrorDuringSetupHandshake(SetupError),
+}
```

Comment:
Shouldn't this not have the error here?
Then instead of the return type of things that return this be something like `SetupHandlerEvent`, it can be `Result<SetupHandlerEvent, SetupError>`. What am I missing?